### PR TITLE
WFLY-14326 Replace 'Java Architecture for XML Binding|JAXB, Java API …

### PR DESCRIPTION
**Pull request by RanabirChakraborty**
Issue: https://issues.redhat.com/browse/WFLY-14326
(Replace "Java Architecture for XML Binding|JAXB, Java API for JSON Binding|JSON-B, Java API for XML-Based Web Services|JAX-WS" references in java doc with corresponding Jakarta EE ones)
Link with: https://github.com/wildfly/wildfly/pull/13808

### CODE REVIEW

**Comment by RanabirChakraborty** [comment]
CC- @asoldano @scottmarlow

File: docs/src/main/asciidoc/_developer-guide/wsconsume.adoc [hunk 1]
```diff
@@ -319,14 +319,14 @@ Also, the following nested elements are supported:
 [cols=",,",options="header"]
 |=================================================
 |Element |Description |Default
-|binding |A JAXWS or JAXB binding file |none
+|binding |A JAXWS or Jakarta XML Binding binding file |none
```
Comment (asoldano) [inline docs/src/main/asciidoc/_developer-guide/wsconsume.adoc:322]: Missing JAXWS replacement

**Review by asoldano — CHANGES_REQUESTED** [review]
@RanabirChakraborty can you please rebase this to resolve conflicts?

**Comment by asoldano** [comment]
@jimma please review

File: docs/src/main/asciidoc/_developer-guide/wsconsume.adoc (hunk 1 above)
Comment (RanabirChakraborty) [reply docs/src/main/asciidoc/_developer-guide/wsconsume.adoc]: Thanks :) Done the changes.

**Review by rsearls — APPROVED** [review]

**Comment by ronsigal** [comment]
Hi @RanabirChakraborty,

It looks fine, but I can find more uses of JAXB, etc. E.g.,
>./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:35:(JAXB)]
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:48:JAXB 2.0
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:94:payload and thus require being compatible with JAXB2. Actually you
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:95:wouldn't need any JAXB annotations for this particular example, because
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:96:JAXB relies on meaningful defaults. For the sake of documentation we put
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:146:http://java.sun.com/webservices/jaxb/[JAXB documentation].
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:214:    -b, --binding=<file>        One or more Jakarta XML Web Services or JAXB binding files
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:251:|ObjectFactory.java |JAXB XML Registry
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:252:|package-info.java |Holder for JAXB package annotations
./docs/src/main/asciidoc/_javaee-guide/Jakarta_XML_Web_Services.adoc:256:Basically _wsconsume_ generates all custom data types (JAXB annotated

I have an old script, rgrep, which does a recursive scan of a directory of files. The first argument is the string your looking for, and the second argument is a template of files you're looking at. For example, if you want to look at every file, use '*'. To get the lines above, I used

> rgrep 'JAXB' '*' |grep -v 'No such file'|grep -v JAXBElement | grep -v JAXBException | grep -v JAXBContext

Here, I'm looking for JAXB in every file. Then I send the list of lines through some filters that remove lines I'm not interested in. "grep -v JAXBElement" will filter out every line that **doesn't** have the Class name JAXBElement, .

**Comment by RanabirChakraborty** [comment]
Hello @ronsigal I'm following this document for changing Java EE specs to Jakarta EE specs https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.3/html-single/development_guide/index#java-ee-specifications-and-the-corresponding-jakarta-ee-specifications_default

Here for `Java Architecture for XML Binding` or `JAXB` the Jakarta one is also `JAXB`.

**Comment by ronsigal** [comment]
Hey @RanabirChakraborty,

I was editing my comment - you're too fast for me. ;-)

It looked like you were replacing every instance of "JAXB" with "Java Architecture for XML Binding", so I just did a brute force search for occurrences of "JAXB". I may well have found instances that didn't need to be changed.

-Ron

**Comment by ronsigal** [comment]
Hmmm, that's strange. I see

Java Architecture for XML Binding | JAXB | JAXB
-- | -- | --
Java API for JSON Binding | JSON-B | Jakarta JSON Binding

so it says to change "Java Architecture for XML Binding" to "JAXB" but to change "Java API for JSON Binding" to "Jakarta JSON Binding". But there's no reference to "Jakarta JSON Binding Specification" ...

**Comment by ronsigal** [comment]
Anyway, I was just thinking that it may be worth doing a mechanical scan to catch loose ends.

**Comment by RanabirChakraborty** [comment]
Sure @ronsigal I'll run the query and follow the steps as you have mentioned.

**Comment by ronsigal** [comment]
I got this for JAX-WS:

>Looking for JAX-WS in files matching *
./.git/objects/pack/pack-45510e776812605434b659c1b4473494e3a5c792.pack
Binary file ./.git/objects/pack/pack-45510e776812605434b659c1b4473494e3a5c792.pack matches
===========
./docs/src/main/asciidoc/_developer-guide/Jakarta_XML_Web_Services_User_Guide.adoc
./docs/src/main/asciidoc/_developer-guide/Jakarta_XML_Web_Services_User_Guide.adoc:5:Services (JAX-WS / JSR-224)] defines the mapping between WSDL and Java
./docs/src/main/asciidoc/_developer-guide/Jakarta_XML_Web_Services_User_Guide.adoc:7:publishing them. JBossWS implements the latest JAX-WS specification,
./docs/src/main/asciidoc/_developer-guide/Jakarta_XML_Web_Services_User_Guide.adoc:659:API for XML-Based Web Services (JAX-WS) 2.2]
===========
./docs/src/main/asciidoc/_developer-guide/wsconsume.adoc
./docs/src/main/asciidoc/_developer-guide/wsconsume.adoc:290:|target |The JAX-WS specification target. Allowed values are 2.0, 2.1
===========
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:50://    //This constructor requires JAX-WS API 2.2. You will need to endorse the 2.2
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:51://    //API jar or re-run wsdl2java with "-frontend jaxws21" to generate JAX-WS 2.1
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:57://    //This constructor requires JAX-WS API 2.2. You will need to endorse the 2.2
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:58://    //API jar or re-run wsdl2java with "-frontend jaxws21" to generate JAX-WS 2.1
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:64://    //This constructor requires JAX-WS API 2.2. You will need to endorse the 2.2
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:65://    //API jar or re-run wsdl2java with "-frontend jaxws21" to generate JAX-WS 2.1
===========
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/cdi/interceptor/CdiInterceptorTestCase.java
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/cdi/interceptor/CdiInterceptorTestCase.java:42: * [JBWS-3441] Support CDI interceptors for POJO JAX-WS services
===========
./testsuite/integration/xts/src/test/java/org/jboss/as/test/txbridge/fromjta/service/FirstServiceATService.java
./testsuite/integration/xts/src/test/java/org/jboss/as/test/txbridge/fromjta/service/FirstServiceATService.java:37: * This class was generated by the JAX-WS RI. JAX-WS RI 2.1.6 in JDK 6 Generated source version: 2.1
===========
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_2.xsd
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_2.xsd:44:        be used for JAX-WS client and JAX-WS endpoint implementations, elements client-config and
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_2.xsd:45:        endpoint-config respectively.  These configurations can include JAX-WS handlers and key/value
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_2.xsd:238:          Defines the name of the JAX-WS handler. The name must be unique within the handler chain.
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_2.xsd:247:          Defines a fully qualified class name for the JAX-WS handler implementation.
===========
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_1.xsd
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_1.xsd:177:          Defines the name of the JAX-WS handler. The name must be unique within the handler chain.
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_1_1.xsd:186:          Defines a fully qualified class name for the JAX-WS handler implementation.
===========
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_2_0.xsd
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_2_0.xsd:44:        be used for JAX-WS client and JAX-WS endpoint implementations, elements client-config and
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_2_0.xsd:45:        endpoint-config respectively.  These configurations can include JAX-WS handlers and key/value
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_2_0.xsd:282:          Defines the name of the JAX-WS handler. The name must be unique within the handler chain.
./webservices/server-integration/src/main/resources/schema/jboss-as-webservices_2_0.xsd:291:          Defines a fully qualified class name for the JAX-WS handler implementation.
===========

**Comment by ronsigal** [comment]
> I'll run the query

Cool. Let me know if you have any trouble. And I'll come back and look again.

**Comment by RanabirChakraborty** [comment]
Hello, @ronsigal I ran the script as well as the command you have provided and found out the left places where `JAX-WS` is mentioned. I have done the changes in a few places. but in most of the places, I think changing will be a bit out of scope.
1. 
```
===========
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:50:// //This constructor requires JAX-WS API 2.2. You will need to endorse the 2.2
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:51:// //API jar or re-run wsdl2java with "-frontend jaxws21" to generate JAX-WS 2.1
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:57:// //This constructor requires JAX-WS API 2.2. You will need to endorse the 2.2
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:58:// //API jar or re-run wsdl2java with "-frontend jaxws21" to generate JAX-WS 2.1
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:64:// //This constructor requires JAX-WS API 2.2. You will need to endorse the 2.2
./testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/wsrm/generated/ReliableService_Service.java:65:// //API jar or re-run wsdl2java with "-frontend jaxws21" to generate JAX-WS 2.1
```
Here I can't replace `JAX-WS API 2.2` with `Jakarta XML Web Services 2.2` because there is nothing like that (https://jakarta.ee/specifications/xml-web-services/)
2. 
```
===========
./webservices/server-integration/src/main/resources/schema/javaee_web_services_1_2.xsd
./testsuite/integration/vdx/target/wildfly/docs/schema/javaee_web_services_1_2.xsd:223:This element also associates a WSDL service with a JAX-WS Provider
./testsuite/integration/vdx/target/wildfly/docs/schema/javaee_web_services_1_2.xsd:269:port components that are JAX-WS Provider implementations.
./testsuite/integration/vdx/target/wildfly/docs/schema/javaee_web_services_1_2.xsd:282:components that are JAX-WS Provider implementations
./testsuite/integration/vdx/target/wildfly/docs/schema/javaee_web_services_1_2.xsd:359:To be used with JAX-WS based runtime only.
./testsuite/integration/vdx/target/wildfly/docs/schema/javaee_web_services_1_2.xsd:671:This is not required when JAX-WS based runtime is used.
```
Here also I can't change the presence of `JAX-WS` because javaee_web_services_1_2.xsd is own by oracle and maybe it'll not be a good idea to replace with Jakarta EE specification names (https://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/javaee_web_services_1_2.xsd).

**Comment by ronsigal** [comment]
Hi @RanabirChakraborty,

> I have done the changes in a few places

Great!

>in most of the places, I think changing will be a bit out of scope.

Sure, it throws a wide net.

Did you also search for the other names / phrases, e.g., JAXB, "Java API for JSON Binding", etc.?

-Ron

**Comment by RanabirChakraborty** [comment]
@ronsigal Yes! Did the same for JAXB and JSON-B :)

**Comment by asoldano** [comment]
@ronsigal can you please confirm if you're fine with this PR contents now? Thanks

**Comment by ronsigal** [comment]
Hi @RanabirChakraborty,

Thank you for doing the extra checks. Looks good to me!

-Ron

**Pull request merged**